core/xml.js

```diff
@@ -596,10 +601,18 @@ Blockly.Xml.domToVariables = function(xmlVariables, workspace) {
     }
     var type = xmlChild.getAttribute('type');
     var id = xmlChild.getAttribute('id');
-    var name = xmlChild.textContent;
-
-    workspace.createVariable(name, type, id);
-  }
+    var name = '';
+    var data = null;
+    for (var i = 0; i < xmlChild.childNodes.length; i++) {
+      if (xmlChild.childNodes[i].nodeType == Blockly.utils.dom.Node.TEXT_NODE) {
```

Comment:
store xmlChild.childNodes[i] in a var, perhaps: ``xmlVariableChild``

Reply:
okay, changed and replaced "i" by"j"